src/Symfony/Component/Form/Extension/Core/Type/BirthdayType.php

```diff
@@ -18,10 +18,19 @@ class BirthdayType extends AbstractType
 {
     public function configureOptions(OptionsResolver $resolver): void
     {
+        $options = $resolver->resolve();
         $resolver->setDefaults([
-            'years' => range((int) date('Y') - 120, date('Y')),
             'invalid_message' => 'Please enter a valid birthdate.',
         ]);
+        if ($options['widget'] === 'single_text') {
+            $resolver->setDefaults([
+                'attr' => ['min' => date('Y-m-d', strtotime('120 years ago')), 'max' => date('Y-m-d')],
```

Comment:
This is the wrong way to do it:

- default values for `attr` is lost as soon as *any* value is passed for `attr`. It is not merged with the default
- you are no longer defining the `years` option (while defining that its only allowed type is an array)
- resolving options inside `configureOptions` is totally broken, and will not take into account the options provided by the user when using that form type to create a form (as they are not available in your call to `resolve()`. We have a way to set smart defaults in the resolver but that's not done by resolving options incorrectly.

To me, you should be using the `buildView` option to add some default attributes in `$view->vars['attr']` based on the configured years instead.

Reply:
Hello @stof ,

Thanks for all those feedback, I'll try to fix it this afternoon